_posts/2024/2024-11-12-Cobbler-v3.3.6-Beginners-Guide.md

```diff
@@ -0,0 +1,560 @@
+
+- **enp0s3** being the *upstream* interface which can access the internet
+
+- **enp0s8** being the *downstream* interface which will server the PXE clients
+
+Create a new **firewalld** ***zone*** which **enp0s8** will operate in:
+
+```shell
+firewall-cmd --permanent --new-zone=cobbler
+firewall-cmd --permanent --zone=cobbler --set-target=DROP
+firewall-cmd --permanent --zone=cobbler --add-service=ssh
+firewall-cmd --permanent --zone=cobbler --add-service=tftp
+firewall-cmd --permanent --zone=cobbler --add-service=http
+firewall-cmd --permanent --zone=cobbler --add-service=https
+firewall-cmd --reload
+```
+
+> Linux ISC DHCP server opens a raw UDP socket with the Linux kernel, bypassing the firewalld rules, so it is not necessary to allow the service/port. Be sure the resulting `/etc/dhcp/dhcpd.conf` file is only configured for the desired interface(s) for hosting DHCP.
+> 
+> - [serverfault](https://serverfault.com/questions/191390/iptables-and-dhcp-questions/) & [stackexchange](https://unix.stackexchange.com/questions/447440/ufw-iptables-not-blocking-dhcp-udp-port-67) sources.
+
+Reconfigure **enp0s8** to operate in the **new firewalld zone**
+
+```shell
+nmcli con mod enp0s8 connection.zone cobbler
+```
+
+> As defined above, this zone will NOT permit inbound ICMP traffic like ping
+
+For simplicity, **enp0s3** can be left in the **default firewalld zone** (verify the current active zones with `firewall-cmd --get-active-zones`)
+
+#### Service confinement
+
+On the topic of network security, we should also restrict our Cobbler services to only listen on the PXE server interface.
+
+- configure HTTP to only listen on the PXE server interface:
+
+```shell
+sed -i 's,^Listen.*,Listen 10.0.0.10:80,' /etc/httpd/conf/httpd.conf
+```
+
+ - Create a local systemd service for tftpd, modifiedying the original config to only listen on a specific IP address:
+ 
+```shell
+systemctl stop tftp.service
+systemctl stop tftp.socket
+
+cp /lib/systemd/system/tftp.service /etc/systemd/system/.
+sed 's,ListenDatagram=69,ListenDatagram=10.0.0.10:69,' /lib/systemd/system/tftp.socket | tee /etc/systemd/system/tftp.socket
+```
+
+- now, reload the systemd configuration and restart services
+
+```shell
+systemctl daemon-reload
+systemctl restart tftp httpd
+```
+
+> Feel free to comment with additional security hardening measures you would employ.
+
+## Tips & Troubleshooting
+
+1. Cobbler's [official docs](https://cobbler.readthedocs.io/en/v3.3.6/installation-guide.html#id2) advise that installing and running Cobbler in a virtual environment is not possible, but I have had no issues related to virtualization.
+
+1. Some dependencies are missing from the official [Cobbler docs](https://cobbler.readthedocs.io/en/v3.3.6/installation-guide.html#packages). Have a look at the [dockerfile](https://github.com/cobbler/cobbler/blob/v3.3.6/docker/rpms/Fedora_34/Fedora34.dockerfile) located in cobbler's source code for a more complete list.
```

Comment:
Cobbler 3.3.7 is around the corner, please file a bug on the backend so we add those deps to the documentation.

Reply:
here's that [ticket](https://github.com/cobbler/cobbler/issues/3845)